### DISCUSSION TITLE
Regarding code capabilities

### DISCUSSION THREAD

**Discussion by Flyingsai** [xcompact3d/Incompact3d#240] [Q&A] [answered]
Respected Sir,

I am learning Xcompact3D. Earlier, I asked a question regarding the capability of code.
Now, I have a different question regarding code capability.

**Background**: I am trying to perform DNS of a translating and rotating cylinder over a _sharp_ flat plate boundary layer.  To simulate the _sharp_  flat plate, I need a small free-slip surface before the flat plate(no-slip surface) starts. Please see the diagram.
(Please notice that my top surface(upper x-z plane) is already a velocity boundary condition such that it provides for a moving and rotating cylinder. )

**Questions:** Is it possible to give two different boundary conditions on the same surface? Here, I want

1. On the bottom surface of the domain(the x-z plane), a free slip boundary condition for some length in the x-direction only and then a no-slip boundary condition (after this free slip). Please see the figure.

My requirements mean that I need to change ncly1=1(for free slip condition) to ncly1=2(no-slip condition) after some x-distance. Is it possible to do this? Or is there another way of meeting my requirements?

![domdoubt](https://github.com/xcompact3d/Incompact3d/assets/50273272/fbcfb92b-b39c-4280-9611-427022bdf2e0)

I am very thankful for your kind replies.
Using Incompact3d, I have already simulated the same problem, but one with a complete no-slip condition on the bottom surface of the domain. The results are kind of similar. But I require a sharp flate plate situation. Please help me with these different boundary condition requirements.

Thanking you
Tejo Sai

**Comment by slaizet** [comment, marked as answer]
Hi Tejo,
Thank you for your message. We have never tried to use different boundary conditions on the same surface. I think it might be possible but you might have some stability issues at the interface between the two types of boundary conditions. As far as the Poisson solver is concerned you would have no change to make as it is designed to work in the same way for ncl=1,2. You would have to make changes for the convective-diffusive terms. I am not quite sure how best to proceed with your problem but it will probably have to add an "i" dependence for the derivative coefficients.
Best wishes,
Sylvain

**Reply by Flyingsai** [reply]
Respected Sir,
Thank you for answering my question. I will look around the code following your instructions. 
Sir, for post-processing the data is there any way to directly write the snapshots in tecplot-supported formats, Instead of the default xdmf format?

**Reply by slaizet** [reply]
I am afraid we do not provide support for Tecplot, only for Paraview.
All the best,
Sylvain
